plugins/search/src/components/SearchFilter/SearchFilter.tsx

```diff
@@ -26,44 +26,53 @@ import {
   FormLabel,
 } from '@material-ui/core';
 
+import {
+  AutocompleteFilter,
+  SearchAutocompleteFilterProps,
+} from './SearchFilter.Autocomplete';
 import { useSearch } from '../SearchContext';
+import { useAsyncFilterValues, useDefaultFilterValue } from './hooks';
 
 const useStyles = makeStyles({
   label: {
     textTransform: 'capitalize',
   },
 });
 
-export type Component = {
+/**
+ * @public
+ */
+export type SearchFilterComponentProps = {
   className?: string;
   name: string;
+  label?: string;
   values?: string[];
+  /**
+   * An async function to return a list of values to be used in the filter. In
+   * the autocomplete filter, the last input value is provided as an input to
+   * allow values to be filtered. This function is debounced and values cached.
+   */
+  asyncValues?: (partial: string) => Promise<string[]>;
```

Comment:
a bit inconvenient perhaps that this is a separate prop, instead of supporting one more type of value to be passed into the regular `values` prop?